C error may happen when using array.any in else-if condition

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 

V full version: V 0.2.2 cb14e42


**OS:** 

OS: macos, macOS, 11.4, 20F71
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
CC version: Apple clang version 12.0.5 (clang-1205.0.22.11)

**What did you do?**


v test tmp_test.v

```
fn test_foo() {
	mut ret := []string{}
	entries := ['a', 'b', 'c']
	if false {
		ret = entries.map(it.capitalize()) // if I comment out this line, test passes.
	} else if entries.any(it == 'a') {
		ret = entries.map(it)
	}
	assert ret == ['a', 'b', 'c']
}
```

**What did you expect to see?**

assertion pass
 
**What did you see instead?**

```
FAIL   447.990 ms /private/tmp/foo_test.v
==================
/tmp/v/test_session_83773/foo_test.11254630459501252592.tmp.c:14693:13: error: use of undeclared identifier '_t3'
        } else if (_t3) {
                   ^
1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4517,7 +4517,10 @@ fn (mut g Gen) if_expr(node ast.IfExpr) {
 		if needs_tmp_var {
 			g.stmts_with_tmp_var(branch.stmts, tmp)
 		} else {
+			// restore if_expr stmt header pos
+			stmt_pos := g.nth_stmt_pos(0)
 			g.stmts(branch.stmts)
+			g.stmt_path_pos << stmt_pos
 		}
 	}
 	g.writeln('}')

--- a/vlib/v/tests/if_expression_test.v
+++ b/vlib/v/tests/if_expression_test.v
@@ -199,3 +199,17 @@ fn min<T>(a T, b T) T {
 fn test_if_expr_with_fn_generic() {
 	assert min(42, 13) == 13
 }
+
+fn test_if_expr_with_complex_array_methods() {
+	mut ret := []string{}
+	entries := ['a', 'b', 'c']
+
+	if false {
+		ret = entries.map(it.capitalize())
+	} else if entries.any(it == 'a') {
+		ret = entries.map(it)
+	}
+
+	println(ret)
+	assert ret == ['a', 'b', 'c']
+}

